src/Microsoft.Scripting/JavaScriptEngineSymbolRegistry.cpp

```diff
@@ -0,0 +1,122 @@
+#include "pch.h"
+#include "JavaScriptEngine.h"
+#include "JavaScriptEngineSymbolRegistry.h"
+#include "JavaScriptFunction.h"
+#include "JavaScriptConverter.h"
+#include "JavaScriptObject.h"
+#include "JavaScriptSymbol.h"
+
+using namespace Platform::Collections;
+using namespace Microsoft::Scripting::JavaScript;
+
+JavaScriptEngineSymbolRegistry::JavaScriptEngineSymbolRegistry(JavaScriptEngine^ engine) :
+    engine_(engine)
+{
+    // Constructor assumes it is invoked from the engine's constructor and the 
+    // engine's context is current on the thread
+    ctor_ = safe_cast<JavaScriptFunction^>(engine_->GetGlobalVariable(L"Symbol"));
+
+    iterator_ = safe_cast<JavaScriptSymbol^>(ctor_->GetPropertyByName(L"iterator"));
+
+    /*match_ = safe_cast<JavaScriptSymbol^>(ctor_->GetPropertyByName(L"match"));
+    replace_ = safe_cast<JavaScriptSymbol^>(ctor_->GetPropertyByName(L"replace"));
+    search_ = safe_cast<JavaScriptSymbol^>(ctor_->GetPropertyByName(L"search"));
+    split_ = safe_cast<JavaScriptSymbol^>(ctor_->GetPropertyByName(L"split"));
+
+    hasInstance_ = safe_cast<JavaScriptSymbol^>(ctor_->GetPropertyByName(L"hasInstance"));
+    isConcatSpreadable_ = safe_cast<JavaScriptSymbol^>(ctor_->GetPropertyByName(L"isConcatSpreadable"));
+    unscopables_ = safe_cast<JavaScriptSymbol^>(ctor_->GetPropertyByName(L"unscopables"));
+    species_ = safe_cast<JavaScriptSymbol^>(ctor_->GetPropertyByName(L"species"));
+    toPrimitive_ = safe_cast<JavaScriptSymbol^>(ctor_->GetPropertyByName(L"toPrimitive"));
+    toStringTag_ = safe_cast<JavaScriptSymbol^>(ctor_->GetPropertyByName(L"toStringTag"));*/
+}
+
+JavaScriptFunction^ JavaScriptEngineSymbolRegistry::Constructor::get()
+{
+    return ctor_;
+}
+
+JavaScriptSymbol^ JavaScriptEngineSymbolRegistry::Iterator::get()
+{
+    return iterator_;
+}
+
+JavaScriptSymbol^ JavaScriptEngineSymbolRegistry::Match::get()
```

Comment:
Should think about removing these properties while they're not supported in Edge.